--- makefile.illumos64	2010-03-30 17:26:26.000000000 +0200
+++ makefile.illumos64	2013-07-14 00:45:54.148195076 +0200
@@ -8,9 +8,9 @@
 # Linux using GCC
 #CXX=g++
 #CXXFLAGS=-O2
-DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
-STRIP=strip
-DESTDIR=/usr
+#DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+#STRIP=strip
+#DESTDIR=/usr
 
 # Linux using LCC
 #CXX=lcc
@@ -46,7 +46,7 @@
 #DEFINES=-D_LARGE_FILES -D_LARGE_FILE_API
 #LIBS=-lbsd
 #STRIP=strip
-#DESTDIR=/usr
+DESTDIR=/usr
 
 # Solaris using CC
 #CXX=CC
@@ -55,12 +55,12 @@
 #STRIP=strip
 #DESTDIR=/usr
 
-# Solaris using GCC (optimized for UltraSPARC 1 CPU)
-#CXX=g++
-#CXXFLAGS=-O3 -mcpu=v9 -mtune=ultrasparc -m32
-#DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
-#STRIP=/usr/ccs/bin/strip
-#DESTDIR=/usr
+# Solaris using GCC (Illumos)
+CXX=g++
+CXXFLAGS=-O3 -m64
+DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DLITTLE_ENDIAN
+STRIP=strip
+#DESTDIR=/usr/local
 
 # Tru64 5.1B using GCC3
 #CXX=g++
@@ -95,7 +95,7 @@
 ##########################
 
 COMPILE=$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(DEFINES)
-LINK=$(CXX)
+LINK=$(CXX) $(CXXFLAGS)
 
 WHAT=UNRAR
 
@@ -136,13 +136,15 @@
 	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
-			install -D unrar $(DESTDIR)/bin/unrar
+			install -D unrar $(DESTDIR)/bin/amd64/unrar
+			install -D unrar.1 $(DESTDIR)/share/man/man1/
 
 uninstall-unrar:
-			rm -f $(DESTDIR)/bin/unrar
+			rm -f $(DESTDIR)/bin/amd64/unrar
+			rm -f $(DESTDIR)/share/man/man1/
 
 install-lib:
-		install libunrar.so $(DESTDIR)/lib
+		install libunrar.so $(DESTDIR)/lib/amd64
 
 uninstall-lib:
-		rm -f $(DESTDIR)/lib/libunrar.so
+		rm -f $(DESTDIR)/lib/amd64/libunrar.so
