--- makefile	2014-04-06 11:08:29.100423459 +0200
+++ makefile.i386	2014-04-06 11:07:43.356814019 +0200
@@ -2,13 +2,13 @@
 # Makefile for UNIX - unrar
 
 # Linux using GCC
-CXX=g++
-CXXFLAGS=-O2
-LIBFLAGS=-fPIC
-DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
-STRIP=strip
-LDFLAGS=-pthread
-DESTDIR=/usr
+#CXX=g++
+#CXXFLAGS=-O2
+#LIBFLAGS=-fPIC
+#DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
+#STRIP=strip
+#LDFLAGS=-pthread
+#DESTDIR=/usr
 
 # Linux using LCC
 #CXX=lcc
@@ -90,6 +90,15 @@
 #LDFLAGS=-static
 #DESTDIR=/usr
 
+# Solaris using GCC (Illumos)
+CXX=g++
+CXXFLAGS=-O3 -m32
+DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DLITTLE_ENDIAN
+STRIP=strip
+#DESTDIR=/usr/local
+
+
+
 ##########################
 
 COMPILE=$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(DEFINES)
@@ -136,10 +145,12 @@
 	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
-			install -D unrar $(DESTDIR)/bin/unrar
+			install -D unrar $(DESTDIR)/bin/i386/unrar
+			install -D unrar.1 $(DESTDIR)/share/man/man1/
 
 uninstall-unrar:
-			rm -f $(DESTDIR)/bin/unrar
+			rm -f $(DESTDIR)/bin/i386/unrar
+			rm -f $(DESTDIR)/share/man/man1/
 
 install-lib:
 		install libunrar.so $(DESTDIR)/lib
