--- makefile	2014-04-06 11:22:27.962873563 +0200
+++ makefile.amd64	2014-04-06 11:22:11.174199139 +0200
@@ -2,13 +2,13 @@
 # Makefile for UNIX - unrar
 
 # Linux using GCC
-CXX=g++
-CXXFLAGS=-O2
-LIBFLAGS=-fPIC
-DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
-STRIP=strip
-LDFLAGS=-pthread
-DESTDIR=/usr
+#CXX=g++
+#CXXFLAGS=-O2
+#LIBFLAGS=-fPIC
+#DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
+#STRIP=strip
+#LDFLAGS=-pthread
+#DESTDIR=/usr
 
 # Linux using LCC
 #CXX=lcc
@@ -90,6 +90,16 @@
 #LDFLAGS=-static
 #DESTDIR=/usr
 
+# Solaris using GCC (Illumos)
+CXX=g++
+CXXFLAGS=-O3 -m64
+LDFLAGS=-m64
+DEFINES=-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DLITTLE_ENDIAN
+STRIP=strip
+#DESTDIR=/usr/local
+
+
+
 ##########################
 
 COMPILE=$(CXX) $(CPPFLAGS) $(CXXFLAGS) $(DEFINES)
@@ -136,13 +146,15 @@
 	$(LINK) -shared -o libunrar.so $(LDFLAGS) $(OBJECTS) $(LIB_OBJ)
 
 install-unrar:
-			install -D unrar $(DESTDIR)/bin/unrar
+			install -D unrar $(DESTDIR)/bin/amd64/unrar
+			install -D unrar.1 $(DESTDIR)/share/man/man1/
 
 uninstall-unrar:
-			rm -f $(DESTDIR)/bin/unrar
+			rm -f $(DESTDIR)/bin/amd64/unrar
+			rm -f $(DESTDIR)/share/man/man1/
 
 install-lib:
-		install libunrar.so $(DESTDIR)/lib
+		install libunrar.so $(DESTDIR)/lib/amd64
 
 uninstall-lib:
-		rm -f $(DESTDIR)/lib/libunrar.so
+		rm -f $(DESTDIR)/lib/amd64/libunrar.so
