diff -Naur lua-5.2.1.orig/src/luaconf.h lua-5.2.1/src/luaconf.h
--- lua-5.2.1.orig/src/luaconf.h	2012-06-16 16:08:08.000000000 -0400
+++ lua-5.2.1/src/luaconf.h	2012-06-16 16:09:25.000000000 -0400
@@ -238,6 +238,7 @@
 ** Compatibility with previous versions
 ** ===================================================================
 */
+#define LUA_COMPAT_ALL
 
 /*
 @@ LUA_COMPAT_ALL controls all compatibility options.

